Day4/src/pipeline/extract_employe_data.py

```diff
@@ -0,0 +1,355 @@
+        CASE 
+        WHEN shift_start_time :: TIMESTAMP :: time BETWEEN '05:00' AND '11:00':: TIME 
+        THEN 'Morning'
+        WHEN shift_start_time :: TIMESTAMP :: time >= '12:00':: TIME
+        THEN 'Evening'
+        ELSE NULL
+        END as shift_type 
+        FROM temporary_punch_details;"""
+    cur.execute(create_query)
+    con.commit()
+
+def create_temporary_tables_for_hours_worked(cur,con):
+    delete="DROP TABLE  IF EXISTS  temporary_hours_table ;"
+    delete2="DROP TABLE  IF EXISTS  temporary_ahours_table;"
+    cur.execute(delete)
+    cur.execute(delete2)
+    con.commit()
+
+    #temporary table for hours worked as paycode =WRK
+    create_query="""
+    CREATE TABLE temporary_hours_table as
+    SELECT DISTINCT(employee_id),punch_apply_date, SUM(CAST(hours_worked as NUMERIC )) as hours_worked
+    FROM raw_timesheet 
+    WHERE paycode ='WRK'
+    GROUP BY employee_id, punch_apply_date,paycode
+    ORDER BY 1 asc, 2 asc;
+    """
+    #temporary table for hours worked as paycode =ABSENT
+    create_query2="""
+    CREATE TABLE temporary_ahours_table as
+    SELECT DISTINCT(employee_id),punch_apply_date, SUM(CAST(hours_worked as NUMERIC )) as hours_worked
+    FROM raw_timesheet 
+    WHERE paycode ='ABSENT'
+    GROUP BY employee_id, punch_apply_date,paycode
+    ORDER BY 1 asc, 2 asc;
+    """
+    cur.execute(create_query)
+    cur.execute(create_query2)
+    con.commit()
+
+def create_table_add_hours(cur,con):
+    delete="DROP TABLE  IF EXISTS  temporary_select_hours_worked ;"
+    delete2="DROP TABLE  IF EXISTS  temporary_select_ahours_worked;"
+    cur.execute(delete)
+    cur.execute(delete2)
+    con.commit()
+    create_query="""
+    CREATE TABLE temporary_select_hours_worked AS
+    SELECT s.employee_id,s.shift_date,s.shift_start_time,s.shift_end_time,s.shift_type,h.hours_worked 
+    FROM temporary_hours_table h
+    INNER JOIN temporary_shift_table s ON h.employee_id =s.employee_id AND h.punch_apply_date=s.shift_date;
+    """
+    create_query2="""
+    CREATE TABLE temporary_select_ahours_worked AS
+    SELECT s.employee_id,s.shift_date,s.shift_start_time,s.shift_end_time,s.shift_type,ah.hours_worked 
+    FROM temporary_ahours_table ah
+    INNER JOIN temporary_shift_table s ON ah.employee_id =s.employee_id and ah.punch_apply_date=s.shift_date;
+    """
+    cur.execute(create_query)
+    cur.execute(create_query2)
+    con.commit()
+
+def insert_data_to_one_table(cur,con):
+    insert_query="""
+    INSERT INTO temporary_select_ahours_worked  
+    SELECT * from  temporary_select_hours_worked;
+    """
+    cur.execute(insert_query)
+    con.commit()
+
+def create_attendance_record(cur,con):
+    delete="DROP TABLE IF EXISTS temporary_attendance"
+    cur.execute(delete)
+    con.commit()
+    create_query="""
+    CREATE TABLE  temporary_attendance AS
+    SELECT DISTINCT(employee_id),   punch_apply_date  ,
+    CASE WHEN paycode='ABSENT' THEN 'ABSENT' ELSE 'PRESENT' END AS attendance
```

Comment:
As I said yesterday, if a person has total hours worked of 5 mins, do you count it as attendance?

Reply:
I had finally stored employee with work hour less than 3 hours as absent